src/main/java/com/microsoft/graph/content/MSBatchRequestContent.java

```diff
@@ -50,12 +52,27 @@ public MSBatchRequestContent() {
 	 * given
 	 */
 	public boolean addBatchRequestStep(final MSBatchRequestStep batchRequestStep) {
-		if (batchRequestStepsHashMap.containsKey(batchRequestStep.getRequestId()))
+		if (batchRequestStepsHashMap.containsKey(batchRequestStep.getRequestId()) ||
+			batchRequestStepsHashMap.size() >= MAX_NUMBER_OF_REQUESTS)
 			return false;
 		batchRequestStepsHashMap.put(batchRequestStep.getRequestId(), batchRequestStep);
 		return true;
 	}
 
+	/**
+	 * Add steps to batch from OkHttp.Request
+	 * @param request the request to add to the batch
+	 * @param arrayOfDependsOnIds ids of steps this step depends on
+	 * @return whether the step was added to the batch or not
+	 */
+	public boolean addBatchRequestStep(final Request request, final String... arrayOfDependsOnIds) {
```

Comment:
If another batch step depends on this one, there is no way to specify (and therefore know) the requestId when using the Request object overload. I suggest that we add an overload that uses the Request object and that takes a requestId so that the customer can make another batch step dependent on this request.

`public boolean addBatchRequestStep(final Request request, final String requestId, final String... arrayOfDependsOnIds)`

Reply:
or I could return the request Id instead of this boolean (and throw ClientExceptions when I can't add the step) to align with what dotnet does

Reply:
update, let me know what you think